main.py

```diff
@@ -72,6 +77,48 @@ def get_smor(req: SmorRequest):
         logger.exception("get_smor threw exception: " + str(e))
         return JSONResponse(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
+"""
+>
+Führungszeugnisse
+Führ<~>ungs<#>zeug<#>nisse<+NN><Fem><Acc><Sg>
+Führ<~>ungs<#>zeug<#>nisse<+NN><Fem><Dat><Sg>
+Führ<~>ungs<#>zeug<#>nisse<+NN><Fem><Gen><Sg>
+Führ<~>ungs<#>zeug<#>nisse<+NN><Fem><Nom><Sg>
+Führ<~>ungs<#>zeugnis<+NN><Neut><Dat><Sg><Old>
+Führ<~>ungs<#>zeugnis<+NN><Neut><Acc><Pl>
+Führ<~>ungs<#>zeugnis<+NN><Neut><Gen><Pl>
+Führ<~>ungs<#>zeugnis<+NN><Neut><Nom><Pl>
+Führung<->s<#>zeug<#>nisse<+NN><Fem><Acc><Sg>
+Führung<->s<#>zeug<#>nisse<+NN><Fem><Dat><Sg>
+Führung<->s<#>zeug<#>nisse<+NN><Fem><Gen><Sg>
+Führung<->s<#>zeug<#>nisse<+NN><Fem><Nom><Sg>
+Führung<->s<#>zeugnis<+NN><Neut><Dat><Sg><Old>
+Führung<->s<#>zeugnis<+NN><Neut><Acc><Pl>
+Führung<->s<#>zeugnis<+NN><Neut><Gen><Pl>
+Führung<->s<#>zeugnis<+NN><Neut><Nom><Pl>
+"""
+@app.post("/lemma", response_model=LemmaResponse)
+def get_lemma(req: SmorRequest):
+    text = req.text
+
+    if not text or text.isspace():
+        logger.error("no content delivered to get_lemma")
+        return JSONResponse(status_code=status-HTTP_400_BAD_REQUEST)
+
+    try:
+        process = Popen(["fst-infl2", SMOR_MODEL], stdin=PIPE, stdout=PIPE)
+        stdout = process.communicate(input=text.encode("utf-8"))[0]
+        #remove morpheme markers
+        stdout = re.sub(r'(<->|<~>|<#>)', '', stdout.decode("utf-8"))
+        #split by the various analyses
+        hypotheses = stdout.split()
+        #remove all the markup that we don't need
+        hypotheses = list(filter(lambda y: '>' not in y, (map(lambda x: re.sub(r'<.*[A-Z].*>', '', x), [h for h in hypotheses]))))
```

Comment:
1. auch hier ist `black` ein Freund, um lange Zeilen code automatisch in besser lesbare, kleinere Blöcke zu formatieren.
2. `[h for h in hypotheses]` iteriert unnötig über `hypotheses`, da `hypotheses` bereits eine `list` of `str` ist - und selbst wenn es keine `list` wäre, könnte `map` auch jedes "iterable" Objekt konsumieren
3. auch das äußere `list` ist nicht notwendig, da `hypotheses` nur noch an `sorted` übergeben wird - `sorted` gibt immer `list` zurück und kann "iterables" konsumieren (auch einen Filter-Iterator)